cpu/native/syscalls.c

```diff
@@ -308,6 +308,10 @@ int fgetc(FILE *fp)
     return getc(fp);
 }
 
+int getchar(void) {
+    return getc(stdin);
```

Comment:
Is this Linux `stdin` or RIOT `stdin` (which might be through a different stdio implementation)?

Reply:
I'm not sure, but either way, through `getc` this gets forwarded to `_native_read`, which runs `if (fd == STDIN_FILENO) return stdio_read(buf, count);`.

So as long as its fileno() is STDIN_FILENO, it's fine.